torchx/lib/torchx/backend.ex

```diff
@@ -150,6 +150,44 @@ defmodule Torchx.Backend do
     NIF.transpose(to_ref(t), 0, 1) |> from_ref(out)
   end
 
+  @impl true
+  def slice(%T{shape: shape} = out, %T{} = t, start_indices, lengths, strides) do
+    ref = to_ref(t)
+
+    for dim <- 0..(length(start_indices) - 1), reduce: ref do
+      ref -> NIF.narrow(ref, dim, Enum.at(start_indices, dim), Enum.at(lengths, dim)) |> unwrap!()
+    end
+    |> NIF.as_strided(shape, steps_to_strides(lengths, strides), 0)
+    |> unwrap!()
+    |> to_tensor(out)
+  end
+
+  def steps_to_strides(shape, steps) do
+    for {dim, step} <- Enum.zip(shape, steps) |> Enum.reverse(), reduce: {1, []} do
+      {offset, strides} -> {offset * dim, [offset * step | strides]}
+    end
+    |> elem(1)
+  end
+
+  ## Aggregators
+
+  @impl true
+  def sum(%T{type: out_type} = out, %T{} = t, opts) do
+    check_out_type!(out_type, "(explicitly cast the input tensor to a signed integer before taking sum)")
+
+    axes = opts[:axes] || []
+    keep_axes = opts[:keep_axes] || false
+
+    NIF.sum(to_ref(t), axes, keep_axes) |> from_ref(out)
+  end
+
+  defp check_out_type!(type, hint) do
+    torch_type(type)
+  rescue
```

Comment:
Avoid rescuing! Instead pass the hint to torch_type and raise the correct thing in the first place. :)